packages/cubejs-server-core/src/core/OrchestratorApi.ts

```diff
@@ -219,6 +220,33 @@ export class OrchestratorApi {
     }
   }
 
+  public async fetchSchema(dataSource: string, securityContext?: { [key: string]: any; }, renew?: boolean) {
+    const cacheDriver = this.orchestrator
+      .getQueryCache()
+      .getCacheDriver();
+
+    const cacheHash = crypto
+      .createHash('md5')
+      .update(JSON.stringify([
+        'FETCH_SCHEMA',
+        dataSource,
+        securityContext
+      ]))
+      .digest('hex');
+
+    if (!renew) {
+      const cachedData = await cacheDriver.get(cacheHash);
+      if (cachedData) {
+        return cachedData;
+      }
+    }
+
+    const driver = await this.driverFactory(dataSource);
+    const tablesSchema = await driver.tablesSchema();
```

Comment:
This will run query bypassing the queue which is not allowed. You can use `BaseDriver#informationSchemaQuery()` method to get the `sql`-query to fetch schemas, and run it as a regular query.